### ISSUE TITLE
Increase the "apparent" trustworthiness of CompatHelper

### ISSUE THREAD

**Issue by fingolfin** [JuliaRegistries/CompatHelper.jl#193]
I've had some people object to using CompatHelper.jl because they were unwilling to install a workflow from a "random" non-official place (no offense intended).

I think it would indeed be easier to convince people to grant access to an action that is in more "official looking" place like <https://github.com/julia-actions> .

(What the actual meaning of that is I can't really say; after all I could go now and register <https://github.com/JuliaActions> and host evil nefarious scripts there... *shrug*)

**Comment by DilumAluthge** [comment]
There is nothing about the `julia-actions` organization that makes it more official or trustworthy than the `bcbi` organization.

In fact, the `bcbi` organization is the official verified GitHub organization of an academic department at a major U.S. university.

Transferring this repository to a different organization is unlikely to happen, as the development of CompatHelper by me was supported by my department, and thus the repository should belong to my department.

That being said, I am very willing to take steps to make CompatHelper more trustworthy and to address any concerns about the trustworthiness of this package.

Can you elaborate on what the trustworthiness concerns are? And then we can brainstorm other ways to address those concerns. Like I said, transferring this repository to a different organization is not an option, but I'm sure that there are other things that we can do.

**Comment by DilumAluthge** [comment]
Just to add: starting in Julia 1.5, when a user does `Pkg.add`, the packages they are installing will not be downloaded from GitHub. Instead, the packages will be downloaded from the official Julia community Pkg Servee hosted at `pkg.julialang.org`, based on the tree hashes provided by the official Julia General registry.

So, starting in Julia 1.5, users will not be downloading CompatHelper (or anything else) from `bcbi`. They will download all packages from the official Julia community Pkg Server. So certainly that should address some concerns.

**Comment by StefanKarpinski** [comment]
I've gone and created the JuliaActions organization. We can use it or not, but now we have it.

**Comment by fingolfin** [comment]
@DilumAluthge first off, let me stress that I find nothing "untrustworthy" about bcbi, CompatHelper or you myself! This is about what some people I work with *perceive* -- I am not actually agreeing with that perception (e.g. you *have* to audit the org/user, you can't just go based on the name, neither in the "trust" nor "distrust" direction). But people are people, and I can understand to a degree why a name like `bcbi`, which sounds like a typical random user name. Like e.g. "fingolfin" ... Tolkien fans can't be serious researchers, can they? You guys at least give your *real* names in your username, so you must be more trustworthy... I mean, anybody with a "FirstnameLastname" style account name must be honest and the name is their actual name, right. Right? Well... ;-)

So, this is obviously a bit silly. But I *can* understand why people are like that; hell, I admit that I also fall for this and have to consciously remind me that this is bad.

And yeah, in theory, I could and should review and audit the code of any package or CI action I use... but in reality, that doesn't happen (and even if I did, one could easily fool me, there is too much to audit; plus, nothing stops one from updating code *after* I reviewed it).

Add in the [leftpad debacle](https://www.theregister.co.uk/2016/03/23/npm_left_pad_chaos/) and its [more recent similar repetition](https://news.ycombinator.com/item?id=22979245). Frankly, I wonder how Julia can avoid running into similar issues.

So I mostly have questions and no grand suggestions, I am afraid.

But one thing that actually might help a bit if the README stressed a bit more what you explained above: that it is developed by a big, respectable institution, which one can check by looking at the org page, which is verified. And ideally "above the fold", not at the very bottom where the NHS acknowledgements are. But even if it was there, it'd help. (That said, one still would have to trust that `bcbi` in turn is responsible and doesn't just let "anybody" add a repo to their org, but, well, we gotta start somewhere).


Finally, you wrote:
> Instead, the packages will be downloaded from the official Julia community Pkg Servee hosted at pkg.julialang.org

Honestly, I don't see how that helps? As long as anybody can trivially register a package and push out arbitrary updates, what's the difference to downloading directly from their homepage? OK, it'd take a few more days to setup initially, but if you want to attract a non-trivial number of "victims", you need that kind of preparation time anyway.


Let me close by thanking you very much for your efforts in making and maintaining this package, and the other things you do for the Julia ecosystem. I apologize if I come across as too negative or ungrateful; this is not my intention.

**Comment by DilumAluthge** [comment]
@fingolfin Thank you for all of your thoughtful input. I really appreciate you opening this issue and providing such a useful discussion.

I've spent a really long time thinking about this. Upon reflection, I think it probably makes the most sense to move CompatHelper to the JuliaRegistries organization. The JuliaRegistries organization is home to the General registry, Registrator.jl, TagBot, and RegistryTools.jl, so I think this makes the most sense.

Since CompatHelper was originally created by BCBI, I have included language in the README as an acknowledgement.
